virtcontainers/sandbox.go

```diff
@@ -1605,8 +1605,7 @@ const maxBlockIndex = 65535
 // getAndSetSandboxBlockIndex retrieves an unused sandbox block index from
 // the BlockIndexMap and marks it as used. This index is used to maintain the
 // index at which a block device is assigned to a container in the sandbox.
-func (s *Sandbox) getAndSetSandboxBlockIndex() (int, error) {
-	var err error
+func (s *Sandbox) getAndSetSandboxBlockIndex() (idx int, err error) {
```

Comment:
- This change looks odd as `idx` is never set. Maybe `currentIndex` should become `idx` in the function body?
- Also, `err` is never set so I don't see how that `defer` ever gets called?

Reply:
@jodh-intel, but wasn't that the case even prior the changes?

What I'm trying to say is, I'm failing to see how the changes are making it worse than it was before.

Reply:
I noticed the second point as well.
Please confirm that the deferred func should be removed.

idx represents the first return value.
It is filled on the return statement.
If the deferred func is removed, this change is not needed.

Reply:
The change to `acrn.go` is fine, but I don't see that the change to this function really helps tbh. In fact as stated, I think it's confusing.

I'd either undo the change to this function, or improve on the change by identfying why this function originally called `defer` by tracing through the git history (maybe it was a copy/paste issue?) and either fixing the logic, or removing the `defer`.

Reply:
I checked commit e9a46580b10a05f9da1d48c5c5490fd981aa318c.
There is no error variable assignment which the deferred func can operate on (before or after that change).

Let me drop the deferred func.